File: models/src/main/scala/traderx/morphir/rulesengine/models/TradeOrder.scala
package traderx.morphir.rulesengine.models

/** Generated based on Morphir.Rulesengine.Models.TradeOrder
*/
object TradeOrder{

  final case class TradeOrder(
    id: morphir.sdk.String.String,
    state: traderx.morphir.rulesengine.models.TradeState.TradeState,
    security: morphir.sdk.String.String,
    quantity: morphir.sdk.Basics.Int,
    accountId: morphir.sdk.Basics.Int,
    side: traderx.morphir.rulesengine.models.TradeSide.TradeSide,
    action: traderx.morphir.rulesengine.models.DesiredAction.DesiredAction,
    filled: morphir.sdk.Maybe.Maybe[morphir.sdk.Basics.Int]
  ){}

}

File: TraderSpec/codebase/target-generated-specfirst/apps/trade-processor/src/main/java/finos/traderx/tradeprocessor/model/TradeOrder.java
package finos.traderx.tradeprocessor.model;

public class TradeOrder {

    public String id;
    private String state;
    private String security;
    private Integer quantity;
    private Integer accountId;
    private TradeSide side;

    public TradeOrder(){}
    
    public TradeOrder(String id, int accountId, String security, TradeSide side, int quantity) {
        this.accountId = accountId;
        this.security = security;
        this.side = side;
        this.quantity = quantity;
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public String getState() {
        return state;
    }

    public Integer getAccountId() {
        return accountId;
    }

    public String getSecurity() {
        return security;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public TradeSide getSide() {
        return side;
    }
}
